server/core/src/main/java/com/redhat/thermostat/server/core/web/CustomResourceConfig.java

```diff
@@ -0,0 +1,24 @@
+package com.redhat.thermostat.server.core.web;
+
+import org.glassfish.jersey.server.ResourceConfig;
+import java.util.Map;
+import com.redhat.thermostat.server.core.internal.web.configuration.ServerConfiguration;
+import io.swagger.jaxrs.config.BeanConfig;
+
+public class CustomResourceConfig extends ResourceConfig {
+    public CustomResourceConfig(Map<String, String> serverConfig) {
+
+        if (serverConfig.containsKey(ServerConfiguration.SWAGGER_UI_ENABLED.toString()) &&
+                serverConfig.get(ServerConfiguration.SWAGGER_UI_ENABLED.toString()).equals("true")) {
+            BeanConfig beanConfig = new BeanConfig();
```

Comment:
Hi!
I tried to put it in  setupResourceHandler, but it didn't get activated (not sure why). From the link you posted, the BeanConfig code is being run inside the constructor of a class that inherits from `Application`. `ResourceConfig` inherits from `Application`, so this is why I decided to create a subclass of ResourceConfig just to execute that code.